httpserver/http_connection.h

```diff
@@ -34,6 +33,8 @@ void make_http_request(boost::asio::io_context& ioc, std::string ip,
                        uint16_t port, std::string target, std::string body,
                        http_callback_t cb);
 
+void request_swarm_update(boost::asio::io_context& ioc, swarm_callback_t cb);
```

Comment:
mh does that even compile? the definition has a `const` in front.
Anyway, better pass the callback by reference to prevent unnecessary copies.
```suggestion
void request_swarm_update(boost::asio::io_context& ioc, const swarm_callback_t& cb);
```